src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -701,5 +684,19 @@ private SeparatedSyntaxList<SwitchExpressionArmSyntax> ParseSwitchExpressionArms
             _pool.Free(arms);
             return result;
         }
+
+        private PatternSyntax ParseListPattern(bool whenIsKeyword)
+        {
+            ParseSubpatternList(
+                out SyntaxToken openBracket,
+                out SeparatedSyntaxList<PatternSyntax> list,
+                out SyntaxToken closeBracket,
+                SyntaxKind.OpenBracketToken,
+                SyntaxKind.CloseBracketToken,
+                static p => p.ParsePattern(Precedence.Conditional));
```

Comment:
Just curious, how did you pick this precedence?

Reply:
Yes, we only use a different precedence in switch arms to avoid consuming `=>` prematurely.